source/client-side-encryption/tests/README.rst

```diff
@@ -1007,4 +1007,57 @@ Drivers that do not support an unlimited maximum pool size MUST skip this test.
       - a find on "db"
    - Expect ``client_keyvault`` to have captured one ``CommandStartedEvent``:
       - a find on "keyvault".
-- ExpectedNumberOfClients: 1
\ No newline at end of file
+- ExpectedNumberOfClients: 1
+
+KMS TLS Tests
+~~~~~~~~~~~~~
+
+.. _ca.pem: https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/x509gen/ca.pem
+
+The following tests that connections to KMS servers with TLS verify peer certificates.
+
+The two tests below make use of mock KMS servers which can be run on Evergreen using `the mock KMS server script <https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/csfle/mock_kms.js>`_.
+
+Setup
+`````
+
+For both tests, do the following:
+
+#. Start a ``mongod`` process with **server version 4.1.9 or later**.
+
+#. Create a ``MongoClient`` (referred to as ``client_encrypted``) for key vault operations with ``keyVaultNamespace`` set to ``keyvault.datakeys``:
+
+Invalid KMS Certificate
+```````````````````````
+
+#. Start a mock KMS server on port 8000 with `ca.pem`_ as a CA file and `expired.pem <https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/x509gen/expired.pem>`_ as a cert file.
+
+#. Call ``client_encrypted.createDataKey()`` with "aws" as the provider and the following masterKey:
+
+   .. code:: javascript
+
+      {
+         "region": "us-east-1",
+         "key": "arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0",
+         "endpoint": "mongodb://127.0.0.1:8000",
+      }
+
+   Expect this to fail with an exception with a message containing the string: "expired".
+
+Invalid Hostname in KMS Certificate
+```````````````````````````````````
+
+#. Start a mock KMS server on port 8000 with `ca.pem`_ as a CA file and `wrong-host.pem <https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/x509gen/wrong-host.pem>`_ as a cert file.
+
+#. Call ``client_encrypted.createDataKey()`` with "aws" as the provider and the following masterKey:
+
+   .. code:: javascript
+
+      {
+         "region": "us-east-1",
+         "key": "arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0",
+         "endpoint": "mongodb://127.0.0.1:8000",
+      }
+
+   Expect this to fail with an exception with a message containing the string: "SANs". The full error message should make
+   a reference to "localhost" not being contained within the list of IP SANs, or server alternative names.
```

Comment:
Let me know if this error substring seems strange. The error message for an incorrect hostname on Ubuntu should be something like "cannot validate certificate for 127.0.0.1 because it doesn't contain any IP SANs".